integration/flink/src/main/java/io/openlineage/flink/visitor/KafkaSinkVisitor.java

```diff
@@ -0,0 +1,64 @@
+package io.openlineage.flink.visitor;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.flink.api.OpenLineageContext;
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.function.Function;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.flink.connector.kafka.sink.KafkaRecordSerializationSchema;
+import org.apache.flink.connector.kafka.sink.KafkaSink;
+
+@Slf4j
+public class KafkaSinkVisitor extends Visitor<OpenLineage.OutputDataset> {
+
+  public KafkaSinkVisitor(@NonNull OpenLineageContext context) {
+    super(context);
+  }
+
+  @Override
+  public boolean isDefinedAt(Object sink) {
+    return sink instanceof KafkaSink;
+  }
+
+  @Override
+  public List<OpenLineage.OutputDataset> apply(Object kafkaSink) {
+    var sink = (KafkaSink<?>) kafkaSink;
+    Field recordSerializerField = FieldUtils.getField(KafkaSink.class, "recordSerializer", true);
+    try {
+      KafkaRecordSerializationSchema<?> serializationSchema =
+          (KafkaRecordSerializationSchema<?>) recordSerializerField.get(kafkaSink);
+      Field topicSelectorField =
+          FieldUtils.getField(
+              serializationSchema.getClass().asSubclass(serializationSchema.getClass()),
+              "topicSelector",
+              true);
+      Field topicSelectorFunctionField =
```

Comment:
Could we create an integration tests which verifies an event emitted by a dummy emitter. 
I am not sure if this is required for this PR or can be done in a next one, but we need test coverage from the early beginning.

Reply:
I'm preparing emiitter (based on Spark emitter), and will create integration test for this case

Reply:
So, will be done in other PR.